### DISCUSSION TITLE
HFS Server failed to rename temp file while continuous upload files

### DISCUSSION THREAD

**Discussion by LinYouMin** [rejetto/hfs#750] [General]
Dear Sirs,

I am running a python script to upload files to HFS server in continuously.
It happened to have 4 files out of 416 files are not able to rename to the correct file name.

Please advise what the possible cause is and how I can resolve this issue.

I'm afraid that the HFS Server console report the error message, it still response the status code of 200.

Thanks!

Scott Lin

# HFS Server console message

(base) youmin@YouMindgdiannao ~ % /Users/youmin/Downloads/hfs-mac-arm64-0/hfs ; exit;
13:16:57 HFS ~ HTTP File Server
13:16:57 © Massimo Melina <a@rejetto.com> - License https://www.gnu.org/licenses/gpl-3.0.txt
13:16:57 started 2024-9-25 1:16:57 PM 
13:16:57 version 0.53.0
13:16:57 build 2024-07-19T14:05:43.478Z
13:16:57 cwd /Users/youmin/.hfs
13:16:57 app /Users/youmin/Downloads/hfs-mac-arm64-0
13:16:57 node v18.5.0
13:16:57 platform darwin arm64
13:16:57 pid 50414
13:16:57 config config.yaml
13:16:57 HINT: type "help" for help
13:16:58 http serving on any network : 80
13:16:58 serving on http://192.168.1.106
13:16:58 serving on http://[2001:b011:7806:df6d:1424:c4a3:4041:47c2]
13:16:58 serving on http://[2001:b011:7806:df6d:75d2:92b7:9f30:5875]
13:16:58 serving on http://[2001:b011:7806:d8f6:48f:9dd6:f6d6:b4b4]
13:16:58 serving on http://[2001:b011:7806:d8f6:b1f2:4577:9b8c:83b7]
13:16:59 loading plugin antibrute
13:19:27 **couldn't rename temp** to /Users/youmin/hfs/PX001/2634/2024-09-25/RJ_Real_2634_20240916_DC3.csv Error: ENOENT: **no such file or directory, rename '/Users/youmin/hfs/PX001/2634/2024-09-25/hfs$upload-RJ_Real_2634_20240916_DC3.csv' -> '/Users/youmin/hfs/PX001/2634/2024-09-25/RJ_Real_2634_20240916_DC3.csv'**
13:19:31 couldn't rename temp to /Users/youmin/hfs/PX001/2634/2024-09-25/RJ_Real_2634_20240906_DC1.csv Error: ENOENT: no such file or directory, rename '/Users/youmin/hfs/PX001/2634/2024-09-25/hfs$upload-RJ_Real_2634_20240906_DC1.csv' -> '/Users/youmin/hfs/PX001/2634/2024-09-25/RJ_Real_2634_20240906_DC1.csv'
13:19:36 couldn't rename temp to /Users/youmin/hfs/PX001/2634/2024-09-25/RJS_2634_20240911_DC1.csv Error: ENOENT: no such file or directory, rename '/Users/youmin/hfs/PX001/2634/2024-09-25/hfs$upload-RJS_2634_20240911_DC1.csv' -> '/Users/youmin/hfs/PX001/2634/2024-09-25/RJS_2634_20240911_DC1.csv'
13:20:17 couldn't rename temp to /Users/youmin/hfs/PX001/2513/2024-09-25/ST_2513_20240915_OW1.csv Error: ENOENT: no such file or directory, rename '/Users/youmin/hfs/PX001/2513/2024-09-25/hfs$upload-ST_2513_20240915_OW1.csv' -> '/Users/youmin/hfs/PX001/2513/2024-09-25/ST_2513_20240915_OW1.csv'

# Python script

def upload_to_hfs(file_path, file_name, hfs_url, username, password):

    # Open the file in binary mode
    with open(file_path, 'rb') as upload_file:
        # Prepare the file for the form-data request
        files = {'file': (file_name, upload_file, 'application/octet-stream')}
        auth = HTTPBasicAuth(username, password)

        # Send the POST request with the file to HFS

        response = requests.post(hfs_url, files=files, auth=auth)

        # Check the server's response
        if response.status_code == 200:
            print("File uploaded successfully!")
            message = "File uploaded successfully!"
            status = True
        else:
            print(f"Failed to upload file. Status code: {response.status_code}")
            print(f"Response: {response.text}")
            message = response.text
            status = False
    
        upload_file.close()

        return status, message

**Comment by rejetto** [comment]
this shouldn't happen, unless an external software is deleting the temp file while it is being uploaded.

can you please provide a zip with your python script and the files to be uploaded, so that i can run it and study the problem?
you can attach it here or send it to a@rejetto.com
